### PR TITLE
Implemented New Plot Events

### PR DESCRIPTION
**Pull request by Kovalskki**
#### Description: 
Adds new plot events

____
#### New Nodes/Commands/ConfigOptions: 

____
#### Relevant Towny Issue ticket:

____
- [x] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

File: src/com/palmergames/bukkit/towny/command/PlotCommand.java [hunk 1]
```diff
@@ -1111,11 +1115,17 @@ public void plotToggle(Player player, TownBlock townBlock, String[] split) {
 					// Prevent plot pvp from being enabled if admin pvp is disabled
 					if (townBlock.getTown().isAdminDisabledPVP() && !townBlock.getPermissions().pvp)
 						throw new TownyException(Translation.of("msg_err_admin_controlled_pvp_prevents_you_from_changing_pvp", "adminDisabledPVP", "on"));
-					
+
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/PlotCommand.java:1118]: revert whitespacing change

File: src/com/palmergames/bukkit/towny/command/PlotCommand.java [hunk 2]
```diff
@@ -1125,20 +1135,40 @@ public void plotToggle(Player player, TownBlock townBlock, String[] split) {
...
 					toggleTest(player, townBlock, StringMgmt.join(split, " "));
+
+					PlotToggleMobsEvent plotToggleMobsEvent= new PlotToggleMobsEvent(townBlock.getTown(), player, choice.orElse(!townBlock.getPermissions().mobs));
+					Bukkit.getPluginManager().callEvent(plotToggleMobsEvent);
+					if (plotToggleMobsEvent.isCancelled()){
+						return;
+					}
+
 					townBlock.getPermissions().mobs = choice.orElse(!townBlock.getPermissions().mobs);
-
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/PlotCommand.java:1141]: revert whitespacing change

File: src/com/palmergames/bukkit/towny/command/PlotCommand.java [hunk 3]
```diff
@@ -1155,7 +1185,7 @@ public void plotToggle(Player player, TownBlock townBlock, String[] split) {
 			} catch (Exception e) {
 				TownyMessaging.sendErrorMsg(player, e.getMessage());
 			}
-			
+
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/PlotCommand.java:1188]: revert whitespacing change

**Review by LlmDl — CHANGES_REQUESTED** [review]
Looks good, just needs a few whitespacing changes reverted.

**Comment by Kovalskki** [comment]
I finally succeeded: D

**Review by LlmDl — APPROVED** [review]

**Comment by LlmDl** [comment]
Do you think you'll be sending in your Jail event fix pr again? Or would you like me to implement that?

**Comment by Kovalskki** [comment]
> Do you think you'll be sending in your Jail event fix pr again? Or would you like me to implement that?
It doesn't matter, you can apply it.

**Comment by LlmDl** [comment]
thanks

**Pull request merged**